File: Backend/Read.er/Models/SaleTrades/SaleTradeReview.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using Read.er.Models.Users;

namespace Read.er.Models.SaleTrades;

/// <summary>
/// Represents a review for a sale trade in the system.
/// </summary>
public class SaleTradeReview
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    [Required]
    public int SellerId { get; set; }
    [JsonIgnore]
    public User Seller { get; set; }

    [Required]
    public int ReviewerId { get; set; }

    [JsonIgnore]
    public User Reviewer { get; set; }

    [Required]
    public int TradeOfferId { get; set; }

    [JsonIgnore]
    public SaleTradeOffer TradeOffer { get; set; }

    [Required]
    [Range(1, 5)]
    public int Rating { get; set; }

    [Required]
    [MaxLength(300, ErrorMessage = "O comprimento do comentário não pode exceder 300 caracteres.")]
    public string Comment { get; set; } = string.Empty;

    public DateTime DateReviewed { get; set; } = DateTime.UtcNow;
}

File: Frontend/reader_frontend_app/lib/models/sale_trade_review.dart
class SaleTradeReview {
  final int id;
  final int sellerId;
  final int reviewerId;
  final int tradeOfferId;
  final int rating;
  final String comment;

  SaleTradeReview({
    required this.id,
    required this.sellerId,
    required this.reviewerId,
    required this.tradeOfferId,
    required this.rating,
    required this.comment,
  });
  factory SaleTradeReview.fromJson(Map<String, dynamic> json) {
    return SaleTradeReview(
      id: json['id'],
      sellerId: json['sellerId'],
      reviewerId: json['reviewerId'],
      tradeOfferId: json['tradeOfferId'],
      rating: json['rating'],
      comment: json['comment'],
    );
  }
}
